footnotes/forms.py

```
@@ -1,15 +1,91 @@
+    )
+    valid_between = GovukDateRangeField(required=False)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.fields[
+            "footnote_type"
+        ].label_from_instance = (
+            lambda obj: f"{obj.footnote_type_id} - {obj.description}"
+        )
+        self.fields["footnote_type"].label = "Footnote type"
+        self.fields["footnote_type"].required = False
+
+        if self.instance:
+            self.fields["code"].disabled = True
+            self.fields["code"].help_text = "You can't edit this"
+            self.fields["code"].initial = self.instance.__str__
+
+            self.fields["footnote_type"].disabled = True
+            self.fields["footnote_type"].help_text = "You can't edit this"
+
+            if self.instance.valid_between.lower:
+                self.fields["start_date"].initial = self.instance.valid_between.lower
+            if self.instance.valid_between.upper:
+                self.fields["end_date"].initial = self.instance.valid_between.upper
+
+        self.helper = FormHelper(self)
+        self.helper.layout = Layout(
+            Field.text(
+                "code",
+                field_width=Fixed.TEN,
+                label_size=Size.SMALL,
+            ),
+            Field("footnote_type", context={"label_size": "govuk-label--s"}),
+            Field("start_date", context={"legend_size": "govuk-label--s"}),
+            Field("end_date", context={"legend_size": "govuk-label--s"}),
+            Submit("submit", "Save"),
+        )
+
+    def clean(self):
+        cleaned_data = super().clean()
+
+        if self.instance and self.instance.footnote_id:
+            cleaned_data["footnote_id"] = self.instance.footnote_id
+
+        # get type from instance if not submitted
+        ctype = cleaned_data.get("footnote_type")
+
+        if not ctype and self.instance and self.instance.footnote_type:
+            ctype = self.instance.footnote_type
```

This code is copied from `CertificateForm`, which raises a validation error if no certificate type was submitted, but you don't do the validation on `footnote_type` so this code does nothing. You should either raise an exception if there is no footnote type or remove this code.